### PR TITLE
Allow multiple NgxSpinner on the same page. Change spinner's options on show call. Showcase of all added features.

### LINKED ISSUE #8: i use one spinner for all pages in my project, how to change options
Hello Napster2210,

could you provide a way to change the spinner options from the ngx-spinner.service.ts ,  i use one element in app.componant.ts  .  
 i show and hide it from any page , but the problem that i need to change some options from other pages.

### LINKED ISSUE #16: How to use multiple ngx-spinner object on single page
How to use multiple ngx-spinner object on single page. how can i show/hide  each one of the using view child

### PR DESCRIPTION
**Pull request by iamavieira**
TL;DR
- Allow Multiple NgxSpinner on the same view. Closes #16 "How to use multiple ngx-spinner object on single page"
- Allow change spinner's options(type, size, bdColor, color) on show call. Closes #8 "i use one spinner for all pages in my project, how to change options"
- Showcase of all added features at ng6-spinner app.
- Add bootstrap to ng6-spinner styles' bundle. To help build the showcase.
- Rename .black-overlay to .overlay; 
- Remove "loading text" input property support(For we now use content projection).

Any questions or suggestions I'm all ears. I hope this helps!

Otherwise, let the commits speak for themselves:

Multiple NgxSpinner on the same view =>
    
    Before, if you wanted to use more than one spinner in the same view, you
    would find yourself out of luck for whenever you tried to activate one,
    all the others would come along.
    
    We needed a way to identify each spinner individually, showing and
    concealing each one of them as we may.
    
    Use cases include save, update, delete or fetch data in multiple places
    at the same time(being fetch and showing data the recurrent case);
    
    Inspired by router-outlet directive implementation I've given our spinners
    a name that works as a constant, not a template expression. That's why I'm
    using @Attribute instead of the traditional @Input(this avoids binding
    change detection to a static string);
    
    A NgxSpinner object was created and all implementation was led by it.
    
    Also I already included it on our public_api, so I could install and test
    the library in our app;
    
    This closes #16 "How to use multiple ngx-spinner object on single page"
    enhancement tagged suggestion.

Change spinner's options on show call =>
    
    Use case is for people that wish to use one single spinner throughout the
    app, passing different options each call, so they can have one single
    instance that behaves as multiple instances.
    
    Given recent improvements on spinner instantiation, that was as easy as
    changing some methods signatures and assigning to the spinner's object
    those options, when they were specified.
    
    Now we can show a spinner with it's name(or blank, in case of unnamed
    spinner), OR it's name and options, that range as long as the Spinner
    interface goes i.e. bdColor, size, color, type.
    
    I decided not to let someone change the spinner's name by options, for it's
    a constant, would be kind of a redundant implementation and... doesn't make
    much sense for the proposed use case.
    
    Note: If you want to change the options for an unnamed spinner it will be
    necessary to call show(undefined, options) explicitly, otherwise just use
    the name of the spinner in place of undefined.
    
    Closes #8 "i use one spinner for all pages in my project, how to change
    options" - Enhancement tag.

Showcase of all added features at ng6-spinner =>
    
    Remember when I said, back in #cc32b4d7bd73f5, that I would make a showcase
    with all changes. So there you have it.
    
    We got multiple spinners, we got one full screen spinner that can have its
    properties changed by passing arguments to show() call(there's a button
    that does that, no need to worry to go to definition and implement
    yourself, though I highly suggest, for you can really understand how it
    works);
    
    And yeah, I think that's that.
    
    It generally helps in clarifying questions w.r.t spinner's implementation.
    
    As soon as this PR gets approved(I hope, it took me quite a while to
    implement all this), I'll personally rewrite the README.MD to match these
    new specifications.

And there are three other minor commits I'm including in this PR: 

Adding bootstrap(Useful for showcase), removing "loading text" input property support(For we now use content projection) and renaming .black-overlay to overlay.

### CODE REVIEW

**Comment by talynone** [comment]
Bump, waiting for this merge so I can switch to this spinner control.

**Comment by engemasa** [comment]
Following - Also interested in this feature

**Comment by engemasa** [comment]
...status?

**Comment by iamavieira** [comment]
@rspl-yuvraj @Napster2210 Glad to here that. I thought you guys were gone.

Anyway, I couldn't help noticing that you've update the project to angular 7, making a myriad of changes to the code base.
Those changes led to conflicts, for my fork was made back on angular 6 support. 
Still, I'm happy to help you solve this issue during the review process.
Just let me know.

Also, I just added one more commit where I improve some code. Some small refactoring.
I had to force push to remove some commits I added to this PR by accident. Sorry about that.
It refers to the "Interceptor implementation" you mentioned in the "Future Plan" section, which this PR already gives support and I'm going to illustrate in a second PR. As soon as I do so, I'll link it here.

Then all Three "Future Plan" enhancements will be done, with this and that next PR I mentioned above.

**Comment by rspl-yuvraj** [comment]
@iamavieira Thanx for your support. I was working on your PR last night, also I have resolved all conflicts just a few things are pending. I'll release these changes in next 2-3 days.

**Comment by rspl-yuvraj** [comment]
Sure @iamavieira I'll push my changes to `Napster2210:feature/PR#44` first, so you can check my changes before release.

**Comment by iamavieira** [comment]
Cool @rspl-yuvraj, thanks for that. 

When you're done, get back to me, so I can revise the final version of the code.

**Comment by Napster2210** [comment]
@iamavieira I've pushed changes to [Napster2210:feature/PR#44](https://github.com/Napster2210/ngx-spinner/tree/feature/PR%2344) branch, you can check now.

**Comment by iamavieira** [comment]
@Napster2210 I just provided some insight on your latest commit on that branch(11d1d154a618315fe7110f459e6556ee9aa5f2ab).
Let me know your thoughts there.

**Comment by iamavieira** [comment]
@rspl-yuvraj I saw your comments [on the latest commit of Napster2210:feature/PR#44there](11d1d154a618315fe7110f459e6556ee9aa5f2ab) on the work I suggested. 
I didn't even get notified.
I just find out that git doesn't consider single lines of comment as code review even though my thoughts provided code review insights.
So I'm basically working for nothing for not even being credited.
Would you mind give me outside collaborator permission, so I can start a review there and get credited for changes made by those insights? Or even create a ngx-spinner team and add the three of us. 
I wish to provide more insights, but without being properly credit for it on that feature branch, it becomes pretty pointless in my opinion.

**Comment by Napster2210** [comment]
@iamavieira Once this PR is merged you'll be credited for your efforts. You'll be a contributor automatically to this package.

**Comment by iamavieira** [comment]
Seems like the perfect time to discuss the state of loading text.
For issue #18 @Napster2210 changed its implementation to use content projection.
But, as issue #47 showed, it's kind of a hassle to reassign or keep multiple variables if you want dynamic text.
@Napster2210 @rspl-yuvraj what about we add this property as an attribute of Spinner and NgxSpinner object, as we're gonna to do for fullScreen, and let people edit it as an option on show call.
But even better. **We keep both**. if the user specify a loading text attribute,  we show it and the content projected. If it does not(via Input binding or show call), we remove that paragraph(<p\>) with *ngif, for it'll be undefined.

That way we can attend either use cases. 
What do you guys think?

**Comment by JoelKap** [comment]
Thanks @iamavieira  for introducing multiple spinners, many thanks, I find it really useful.

**Comment by iamavieira** [comment]
you`re welcome joel.
glad to help!
